ui/app/components/calendar-widget.js

```diff
@@ -3,139 +3,115 @@ import layout from '../templates/components/calendar-widget';
 import { setComponentTemplate } from '@ember/component';
 import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
-
+import { ARRAY_OF_MONTHS, parseAPITimestamp } from 'core/utils/date-formatters';
+import { addYears, isSameYear, subYears } from 'date-fns';
 /**
  * @module CalendarWidget
- * CalendarWidget components are used in the client counts metrics. It helps users understand the ranges they can select.
+ * CalendarWidget component is used in the client counts dashboard to select a month/year to query the /activity endpoint.
+ * The component returns an object with selected date info, example: { dateType: 'endDate', monthIdx: 0, monthName: 'January', year: 2022 }
  *
  * @example
  * ```js
- * <CalendarWidget
- * @param {array} arrayOfMonths - An array of all the months that the calendar widget iterates through.
- * @param {string} endTimeDisplay - The formatted display value of the endTime. Ex: January 2022.
- * @param {array} endTimeFromResponse - The value returned on the counters/activity endpoint, which shows the true endTime not the selected one, which can be different. Ex: ['2022', 0]
- * @param {function} handleClientActivityQuery - a function passed from parent. This component sends the month and year to the parent via this method which then calculates the new data.
- * @param {function} handleCurrentBillingPeriod - a function passed from parent. This component makes the parent aware that the user selected Current billing period and it handles resetting the data.
- * @param {string} startTimeDisplay - The formatted display value of the endTime. Ex: January 2022. This component is only responsible for modifying the endTime which is sends to the parent to make the network request.
- * />
+ * <CalendarWidget @startTimestamp={{this.startTime}} @endTimestamp={{this.endTime}} @selectMonth={{this.handleSelection}} />
  *
+ *    @param {string} startTimestamp - ISO timestamp string of the calendar widget's start time, displays in dropdown trigger
+ *    @param {string} endTimestamp - ISO timestamp string for the calendar widget's end time, displays in dropdown trigger
+ *    @param {function} selectMonth - callback function from parent - fires when selecting a month or clicking "Current billing period"
+ *  />
```

Comment:
Nice work! It looks like the `CalendarWidget` was originally developed specifically for client counts and now with this simplified API it can be used anywhere!